### ISSUE TITLE
forms: Support negative indexes in `FormArray`

### ISSUE THREAD

**Issue by amitbeck** [angular/angular#44642]
### Which @angular/* package(s) are relevant/releated to the feature request?

forms

### Description

Nowadays attempting to retrieve a control in a negative index returns `undefined` (due to underlying usage of native `Array`).
I would like `FormArray` to support relative indexing. Methods like `at()` and `removeAt()` could support negative integers as indexes, which count back from the last item in the array.

### Proposed solution

Similar to the ES2021 [`Array.prototype.at()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at), support negative integers as indexes:
```ts
 // the following are equivalent:
formArray.at(-1)
formArray.at(formArray.length - 1)
```

Methods which should support relative indexing:
1. `at()`
2. `removeAt()`
3. `insert()`
4. `setControl()`

See proposed solution implementation in https://github.com/angular/angular/pull/44631

### Alternatives considered

One could use `formArray.at(formArray.length - 1)` like in indices of native `Array`s pre-ES2021, but this can be exhausting and causes longer lines of code.

**Comment by DavidHazaniInt** [comment]
Good idea!

**Comment by oran-intsights** [comment]
Very Important!

**Comment by AndrewKushnir** [comment]
@amitbeck thanks for creating this feature request. According to the Feature Request process ([documented here](https://github.com/angular/angular/blob/e29eb8abfbfdfb793e08ac88e14b7ba6d2ff2136/docs/GITHUB_PROCESS.md#feature-request-process)), this ticket reached 20 upvotes and we've discussed it with the Angular team.

Given the fact that:
- `FormArray` class represents an Array data structure and using negative indices is common in array methods (such as `splice` and `at`)
- This feature would have minimal impact on the bundle size (we would even be able to pass the index as is in case of `at` function)

We'd like to proceed with this feature request. I'll reopen PR #44631 and provide some feedback there.

Thank you.

**Comment by jd-apprentice** [comment]
I'll love to have this!

**Comment by amitbeck** [comment]
> I'll love to have this!

@jd-apprentice It's been implemented and already merged! :tada:

**Comment by amitbeck** [comment]
@dylhunn feel free to close this issue at it's resolved by f490c2de4e1ada26e62e35c0d70cb90e55297658

**Comment by AndrewKushnir** [comment]
@amitbeck thanks for contributing to Angular 👍
